src/lib/support/CHIPMem-Malloc.cpp

```diff
@@ -100,30 +105,54 @@ void MemoryAllocatorShutdown()
 #endif // CHIP_CONFIG_MEMORY_DEBUG_DMALLOC
 }
 
+extern "C" {
+
+WEAK void * __wrap_malloc(size_t size)
+{
+    return malloc(size);
+}
+
+WEAK void __wrap_free(void * ptr)
+{
+    free(ptr);
+}
+
+WEAK void * __wrap_calloc(size_t num, size_t size)
+{
+    return calloc(num, size);
+}
+
+WEAK void * __wrap_realloc(void * ptr, size_t new_size)
+{
+    return realloc(ptr, new_size);
+}
+
+} // extern "C"
+
```

Comment:
There is no need for this code: without linker wrapping, there is already these calls.

Reply:
Fixed